File: app/src/main/java/wjy/yo/ereader/entity/book/Book.java
package wjy.yo.ereader.entity.book;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Ignore;

import java.util.Date;

import wjy.yo.ereader.entity.FetchedData;
import wjy.yo.ereader.entity.anno.AnnoFamily;
import wjy.yo.ereader.entity.userdata.UserBook;


@Entity(tableName = "book")
public class Book extends FetchedData {

    private String code;

    private String name;

    private String zhName;

    private String author;

    private String zhAuthor;

    private String tags;

    @ForeignKey(entity = AnnoFamily.class, parentColumns = "id", childColumns = "annotationFamilyId")
    private String annotationFamilyId;

    private Date chapsLastFetchAt;

    @Ignore
    private UserBook userBook;

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getZhName() {
        return zhName;
    }

    public void setZhName(String zhName) {
        this.zhName = zhName;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getZhAuthor() {
        return zhAuthor;
    }

    public void setZhAuthor(String zhAuthor) {
        this.zhAuthor = zhAuthor;
    }

    public String getTags() {
        return tags;
    }

    public void setTags(String tags) {
        this.tags = tags;
    }

    public String getAnnotationFamilyId() {
        return annotationFamilyId;
    }

    public void setAnnotationFamilyId(String annotationFamilyId) {
        this.annotationFamilyId = annotationFamilyId;
    }

    public Date getChapsLastFetchAt() {
        return chapsLastFetchAt;
    }

    public void setChapsLastFetchAt(Date chapsLastFetchAt) {
        this.chapsLastFetchAt = chapsLastFetchAt;
    }

    public UserBook getUserBook() {
        return userBook;
    }

    public void setUserBook(UserBook userBook) {
        this.userBook = userBook;
    }

    @Override
    public String toString() {
        return id + " " + name;
    }
}

File: app/src/main/java/wjy/yo/ereader/entity/book/Book.kt
package wjy.yo.ereader.entity.book

import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.Ignore

import java.util.Date

import wjy.yo.ereader.entity.FetchedData
import wjy.yo.ereader.entity.anno.AnnoFamily
import wjy.yo.ereader.entity.userdata.UserBook


@Entity(tableName = "book")
open class Book : FetchedData("") {

    var code: String? = null

    var name: String? = null

    var zhName: String? = null

    var author: String? = null

    var zhAuthor: String? = null

    var tags: String? = null

    @ForeignKey(entity = AnnoFamily::class, parentColumns = arrayOf("id"), childColumns = arrayOf("annotationFamilyId"))
    var annotationFamilyId: String? = null

    var chapsLastFetchAt: Date? = null

    @Ignore
    var userBook: UserBook? = null

    override fun toString(): String {
        return "$id $name"
    }
}
